src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
 
     /*********************************************************************************************************
     @description
+        Update Contact's mailing address (with State & Country Picklists enabled) and mark it as override,
+        to cause creation of a new non-default Address record with the State & Country values set as expected.
+    verify:
+        Contact gets new address
+        HH sticks with default Address
+        New Address created with State value set as expected
+    **********************************************************************************************************/
+    static testMethod void testContactAddressOverrideForStateChangeWithStateAndCountryPicklistEnabled() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testContactAddressMarkOverrideWithStateAndCountryPicklistsEnabled') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+        System.assertEquals(1, getAddresses().size());
+
+        // Modify the Contact addresses directly, and set override
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.put('MailingCountryCode', 'US');
```

Comment:
Should we test changing the State only as it is the test mentioned in the issue?

Reply:
Yes, that works well with the new implementation.